src/api/sign/lock.js

```diff
@@ -0,0 +1,28 @@
+const P = require('bluebird')
+const { post } = require('request-promise')
+
+module.exports = async (efx, wrapperAddress, duration = 25) => {
+  const {web3, config} = efx
+
+  const getBlockNumber = P.promisify(web3.eth.getBlockNumber)
+
+  const blockNumber = (await getBlockNumber()) + duration
+
+  // POST request payload
+  const payload = {
+    address: config.account, // your ethereum address
+    tokenAddress: wrapperAddress,
+    unlockUntil: blockNumber
+  }
+
+  const url = config.api // do we add anything to this path?
+
```

Comment:
@hems think this is the wrong way around for locking (and payload and request is just for unlocking)?

Reply:
@plutoegg indeed for locking the user will interact directly with web3, this is already updated on the latest version which i'm about to push.